### PR TITLE
feat: added tokenizer params to the listing

### LINKED ISSUE #54: Extend model listing API with limits (tokenize, rate endpoint)
### All optionals

* **prompt_token_unit** - (examples: gpt-3.5-turbo-0314, gpt-3.5-turbo-0613, chars (utf-8) and ...) defines the strategy for counting the number of tokens.
* **max_prompt_messages** - is the max number of messages that the assistant is guaranteed to accept.
* **max_system_messages** - is the max number of system messages that the assistant is guaranteed to accept.
* **tokenizer_endpoint** - url to API that can tokenize input prompt. (add to open api website doc)
* **rate_endpoint** -  #32

### PR DESCRIPTION
**Pull request by adubovik**
Added params related to tokenization to the listing as part of #54 

DIAL core config:
* `features.tokenizeEndpoint` (URL optional) - the endpoint which allows to tokenize prompt 
* `features.truncatePromptEndpoint` (URL optional) - the endpoint which serves context trimming API call. It trims the chat history up to the point when it fit requested number of tokens specified in request field `max_prompt_tokens`.
* `tokenizerModel` (string optional) - the reference model whose tokenization algorithm is matching the tokenization algorithm of the given model. It allows a user to make tokenization on their side. It's possible for models whose tokenization algorithm is publically known and implemented in a SDK (e.g. [GPT](https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb) and [Anthropic](https://github.com/anthropics/anthropic-sdk-python/blob/main/src/anthropic/_tokenizers.py)). Models fall into families which share the same tokenization algorithm. `tokenizer.referenceModel` basically points to a representative model from a corresponding family. As a rule of thumb, choose the oldest representative from a family. See the families below.

Listing:
* `features.tokenize` (boolean optional) - `true` means that the core has `<server host>/v1/deployments/<deployment name>/tokenize` endpoint
* `features.truncate_prompt` (boolean optional) - `true` means that the core has `<server host>/v1/deployments/<deployment name>/truncate_prompt` endpoint
* `tokenizer_model` (string optional) - same value as in the core config

<details>
  <summary>Model families</summary>
  
```yaml
_tokenization_families:
  - gpt:
    - family_1:
      - gpt-3.5-turbo-0613
      - gpt-3.5-turbo-16k-0613
      - gpt-4-0314
      - gpt-4-32k-0314
      - gpt-4-0613
      - gpt-4-32k-0613
      # tokens_per_message = 3, tokens_per_name = 1
    - family_2:
      - gpt-3.5-turbo-0301
      # tokens_per_message=4, tokens_per_name=-1
    - family_3:
      - text-embedding-ada-002
      # just a string
  - gpt_encoding: cl100k_base
  - gpt_refs:
    - https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb
    - https://platform.openai.com/docs/models
    - https://tiktokenizer.vercel.app/
  - palm:
    - family_1:
      - chat-bison@001
      - codechat-bison@001
      - textembedding-gecko@001
  - anthropic:
    - family_1:
      - anthropic.claude-instant-v1
      - anthropic.claude-v1
      - anthropic.claude-v2
  - anthropic_refs:
    - https://github.com/anthropics/anthropic-sdk-python/blob/main/src/anthropic/_tokenizers.py
```
</details>

### CODE REVIEW

**Review by artsiomkorzun — DISMISSED** [review]

File: src/main/java/com/epam/aidial/core/controller/ControllerSelector.java [hunk 1]
```diff
@@ -157,6 +158,13 @@ private static Controller selectPost(Proxy proxy, ProxyContext context, String p
             return () -> controller.handle(deploymentId);
         }
 
+        match = match(PATTERN_TOKENIZE_RESPONSE, path);
```
Comment (astsiapanay) [inline src/main/java/com/epam/aidial/core/controller/ControllerSelector.java]: Please add Unit test to TestControllerSelector

File: src/main/java/com/epam/aidial/core/controller/TokenizeController.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.URL;
+import java.util.Optional;
+
+@Slf4j
+@RequiredArgsConstructor
+public class TokenizeController {
```
Comment (astsiapanay) [inline src/main/java/com/epam/aidial/core/controller/TokenizeController.java]: Please add Unit tests

File: src/main/java/com/epam/aidial/core/controller/ControllerSelector.java (hunk 1 above)
Comment (adubovik) [reply src/main/java/com/epam/aidial/core/controller/ControllerSelector.java]: Added tests

File: src/main/java/com/epam/aidial/core/controller/TokenizeController.java (hunk 2 above)
Comment (adubovik) [reply src/main/java/com/epam/aidial/core/controller/TokenizeController.java]: Added

**Review by astsiapanay — APPROVED** [review]

**Pull request merged**